### ISSUE TITLE
Create process to update OakPAL checklist

### ISSUE THREAD

**Issue by justinedelson** [Adobe-Consulting-Services/acs-aem-commons#1872]
Pulling up from comment in https://github.com/Adobe-Consulting-Services/acs-aem-commons/pull/1566#issuecomment-487933524

We should definitely have a way of generating this automatically.

@adamcin can you advise?

**Comment by kwin** [comment]
This is not only about the meta data of forcedRoots but also about the node type definitions (cnd files). Both tend to change with each AEM version (the latter is much more stable though)... I created  https://github.com/adamcin/oakpal/issues/9 for that because I am not sure this really belongs to acs-aem-commons....

**Comment by kwin** [comment]
With AEM 6.5 I no longer get any content classification violation in the built-in health check so most probably we could also get rid of the exclusions from https://github.com/Adobe-Consulting-Services/acs-aem-commons/blob/88a9bec43adc3f39516e500a55e3d46e315739b6/oakpal-checks/src/main/resources/OAKPAL-INF/checklist/acs-internal.json#L34 once we have updated the `forcedRoots`

**Comment by adamcin** [comment]
For the CND generation, I think I can make a simple plugin for recursively scanning xml files in a jcr_root directory for `jcr:primaryType` and `jcr:mixinTypes` attributes, which would then call a servlet to generate a CND for just those types, or maybe I can even do it using a path to a directory on the filesystem, which would make it less AEM/OSGi-specific. 

For the Forced Root generation for the granite content usage annotations, I think I originally used a custom JSP, but maybe I could also do this using filesystem access to the repository, and generalize it to execute a JCR query provided by a parameter, to generate the forcedRoots JSON stream with primaryType and mixinTypes.

**Comment by adamcin** [comment]
I am currently working on adding this tooling to oakpal, BTW.

**Comment by adamcin** [comment]
In 1.3.0-RC2, I have added a feature called Webster to the oakpal framework, with the primary interface being an `oakpal:webster` maven mojo that is executed in a module directory, but not as part of the normal build lifecycle. Given a `webster.segmentStore` parameter that references an absolute path to a local oak repository's `segmentstore` directory, it uses a read-only CompositeNodeStore mount to construct a temporary JCR repository that imports the namespaces, privileges, and nodetypes from the source repository, and that at least exposes all of the content to query-by-traversal, if not by index. The plugin descriptor in the pom must specify one or more websterTargets in the configuration to define the path and type of file to be generated. Generally speaking, filevault-package modules will specify `nodetypes` and `privileges` targets, where as checklist jar modules will specify `checklist` targets. Here are some important features:
1. Package nodetypes.cnd and privileges.xml are generated to automatically include only those definitions that are referenced by .content.xml and *.cnd files in the filevault archive root. Very little configuration appears to be needed for maintaining these files at this point, so I am willing to conclude that for anyone other than Adobe to publish authoritative artifacts for these files would likely be overkill, and may introduce more issues by exporting more definitions than necessary. 
2. I have added a `net.adamcin.oakpal.core.JsonCnd`-managed structure with the root-level key, `jcrNodetypes` to the checklist json format to replace the now deprecated `cndNames` key. Combined with the existing `jcrNamespaces` list of prefix-to-uri mappings, this allows the JCR dependencies in a checklist to be managed by webster in a canonical, self-contained form, with the goal of making git diffs as readable as possible, and reducing the impact of drifting external dependencies.
